src/nucname.cpp

```diff
@@ -969,6 +969,79 @@ int pyne::nucname::mcnp_to_id(std::string nuc) {
   return mcnp_to_id(pyne::to_int(nuc));
 }
 
+/************************/
+/*** openmc functions ***/
+/************************/
+std::string pyne::nucname::openmc(int nuc) {
+  std::string nucname = name(nuc);
+
+  // check aaa value
+  int aaa = anum(nuc);
+  if (aaa == 0) {
+    nucname.append("0");
+  }
+  // format metadata
+  if ('M' == nucname.back()) {
+    nucname.back() = '_';
+    nucname.append("m");
+    int meta_id = snum(nuc);
+    std::string meta_str = std::to_string(meta_id);
+    nucname.append(meta_str);
+  }
+  return nucname;
+}
+
+std::string pyne::nucname::openmc(const char * nuc) {
+  std::string newnuc (nuc);
+  return openmc(newnuc);
+}
+
+std::string pyne::nucname::openmc(std::string nuc) {
+  return openmc(id(nuc));
+}
+
+//
+// OPENMC -> id
+//
+int pyne::nucname::openmc_to_id(const char * nuc) {
+  return openmc_to_id(std::string(nuc));
+}
+
+int pyne::nucname::openmc_to_id(std::string nuc) {
+  std::string nucname;
```

Comment:
Are there ways to use other `*_to_id` functions to bootstrap this?  Is the OpenMC format similar enough to others that you can tweak the string and then call the other function???

Reply:
I think the thing to do would be to add support for GND format in the general `id()` function. There are already a few string-based formats included and I worry about clashing with them. Ideally that function would be a series of functions that attempt to read the format and generate a PyNE id. I don't know if I want to take that on though...